### PR TITLE
fix: only show new versions of spaces if they are newer than the local copies

### PR DESCRIPTION
**Pull request by hiporox**

### CODE REVIEW

File: src/common/data/stores/app/homebase/homebaseStore.ts [hunk 1]
```diff
@@ -59,6 +60,16 @@ export const createHomeBaseStoreFunc = (
       const spaceConfig = JSON.parse(
         await get().account.decryptEncryptedSignedFile(fileData),
       ) as SpaceConfig;
+      const currentHomebase = get().homebase.homebaseConfig;
+      if (
+        spaceConfig.timestamp &&
```
Comment (ryannorton) [inline src/common/data/stores/app/homebase/homebaseStore.ts]: Handle undefined `spaceConfig`

File: src/common/data/stores/app/space/spaceStore.ts [hunk 2]
```diff
@@ -129,6 +129,19 @@ export const createSpaceStoreFunc = (
       const spaceConfig = JSON.parse(
         await get().account.decryptEncryptedSignedFile(fileData),
       ) as DatabaseWritableSpaceConfig;
+      const currentLocalCopy = get().space.localSpaces[spaceId];
+      if (
+        spaceConfig &&
```
Comment (ryannorton) [inline src/common/data/stores/app/space/spaceStore.ts:134]: Handle undefined `spaceConfig`/`currentLocalCopy`

**Review by ryannorton — CHANGES_REQUESTED** [review]

File: src/common/data/stores/app/homebase/homebaseStore.ts [hunk 3]
```diff
@@ -59,6 +60,16 @@ export const createHomeBaseStoreFunc = (
...
+        currentHomebase &&
+        currentHomebase.timestamp &&
```
Comment (ryannorton) [inline src/common/data/stores/app/homebase/homebaseStore.ts]: Should also return true if local config has a timestamp and the fetched one doesn't to fix this issue on existing users' next save

File: src/common/data/stores/app/space/spaceStore.ts [hunk 4]
```diff
@@ -129,6 +129,19 @@ export const createSpaceStoreFunc = (
...
+        spaceConfig.timestamp &&
```
Comment (ryannorton) [inline src/common/data/stores/app/space/spaceStore.ts]: Should also return true if local config has a timestamp and the fetched one doesn't

**Comment by hiporox** [comment]
@ryannorton Fixed comments

**Pull request merged**